commit 2cb9317377caaec647d7485bc53ab33a0b54f27c
Author: Sean Christopherson <seanjc@google.com>
Date:   Mon Feb 27 14:10:10 2023 +0530

    KVM: x86: Raise an event request when processing NMIs if an NMI is pending
    
    Don't raise KVM_REQ_EVENT if no NMIs are pending at the end of
    process_nmi().  Finishing process_nmi() without a pending NMI will become
    much more likely when KVM gains support for AMD's vNMI, which allows
    pending vNMIs in hardware, i.e. doesn't require explicit injection.
    
    Signed-off-by: Santosh Shukla <Santosh.Shukla@amd.com>
    Link: https://lore.kernel.org/r/20230227084016.3368-6-santosh.shukla@amd.com
    Signed-off-by: Sean Christopherson <seanjc@google.com>

diff --git a/arch/x86/kvm/x86.c b/arch/x86/kvm/x86.c
index 237c483b1230..3c995483b405 100644
--- a/arch/x86/kvm/x86.c
+++ b/arch/x86/kvm/x86.c
@@ -10150,7 +10150,9 @@ static void process_nmi(struct kvm_vcpu *vcpu)
 
 	vcpu->arch.nmi_pending += atomic_xchg(&vcpu->arch.nmi_queued, 0);
 	vcpu->arch.nmi_pending = min(vcpu->arch.nmi_pending, limit);
-	kvm_make_request(KVM_REQ_EVENT, vcpu);
+
+	if (vcpu->arch.nmi_pending)
+		kvm_make_request(KVM_REQ_EVENT, vcpu);
 }
 
 void kvm_make_scan_ioapic_request_mask(struct kvm *kvm,